src/main/java/com/commercetools/sync/products/ProductSync.java

```diff
@@ -181,11 +181,16 @@ private void processFetchedProducts(@Nonnull final Set<Product> fetchedProducts)
     }
 
     @Nonnull
-    private CompletionStage<List<Optional<Product>>> createOrUpdateProducts() {
-        return productService.createProducts(draftsToCreate)
-                             .thenAccept(createdProducts ->
-                                 updateStatistics(createdProducts, draftsToCreate.size()))
-                             .thenCompose(ignoredResult -> syncProducts(productsToSync));
+    private CompletionStage<Void> createOrUpdateProducts() {
```

Comment:
I'd say it is <code>create<b>And</b>UpdateProducts</code>, since there is no dependency between these actions.
Actually, because of this name i was confused during reviewing: first i thought the stages should be chained.

Reply:
Actually, the reason why it's `OR` and not `AND` in the name, is because it could be that there are no creates and updates only, and it also could be that there are creates only and no updates. Hmm, but I see your point of confusion. Hmm, I will change to `AND` since it's at least less confusing.

Reply:
d963190